lib/Doctrine/DBAL/Driver/Connection.php

```diff
@@ -15,62 +14,80 @@
 {
     /**
      * Prepares a statement for execution and returns a Statement object.
+     *
+     * @param string $sql The SQL query to prepare.
+     *
+     * @return Statement The prepared statement.
      */
     public function prepare(string $sql) : Statement;
 
     /**
      * Executes an SQL statement, returning a result set as a Statement object.
      *
+     * @param string $sql The SQL query to execute.
+     *
+     * @return ResultStatement The result statement.
+     *
      * @throws DBALException
      */
     public function query(string $sql) : ResultStatement;
 
     /**
      * Quotes a string for use in a query.
      *
-     * @param mixed $input
-     * @param int   $type
+     * @param string $input The parameter to quote.
      *
-     * @return mixed
+     * @return string The quoted string.
      */
-    public function quote($input, $type = ParameterType::STRING);
+    public function quote(string $input) : string;
 
     /**
      * Executes an SQL statement and return the number of affected rows.
      *
+     * @param string $statement The SQL query to execute.
+     *
+     * @return int The number of affected rows.
+     *
      * @throws DBALException
      */
     public function exec(string $statement) : int;
 
     /**
      * Returns the ID of the last inserted row or sequence value.
      *
-     * @param string|null $name
+     * If a sequence name was not specified, lastInsertId() returns a string representing the row ID of the last row
+     * that was inserted into the database.
+     *
+     * If a sequence name was specified, lastInsertId() returns a string representing the last value retrieved from the
+     * specified sequence object.
      *
-     * @return string
+     * This method throws a DriverException if a value cannot be returned, in particular when:
+     *
+     * - no sequence name was provided, but the platform requires one, or the last statement did not return an identity;
+     * - a sequence name was provided, but the platform does not support sequences, or no such sequence exists.
```

Comment:
I'd make sense to split the parts separated by "or" into different items.

Reply:
Some drivers (PDO OCI) don't support this API at all. In this case, we'll throw an exception as well.

Reply:
I further clarified the docblocks, with 5 bullet points.